Packs/GitGuardian/Integrations/GitGuardianEventCollector/GitGuardianEventCollector.py

```diff
@@ -0,0 +1,383 @@
+import demistomock as demisto
+from CommonServerPython import *
+import urllib3
+from typing import Any
+
+# Disable insecure warnings
+urllib3.disable_warnings()
+
+""" CONSTANTS """
+
+DATE_FORMAT = "%Y-%m-%dT%H:%M:%S.%fZ"
+VENDOR = "gitguardian"
+PRODUCT = "enterprise"
+DEFAULT_PAGE_SIZE = 1000
+EVENT_TYPE_TO_TIME_MAPPING = {
+    "audit_log": "gg_created_at",
+    "incident": "last_occurrence_date",
+}
+EVENT_TYPE_TO_ENDPOINT = {
+    "audit_log": "/audit_logs",
+    "incident": "/secrets",
+}
+
+""" CLIENT CLASS """
+
+
+class Client(BaseClient):
+    """Client class to interact with the service API
+
+    This Client implements API calls, and does not contain any Demisto logic.
+    Should only do requests and return data.
+    It inherits from BaseClient defined in CommonServer Python.
+    Most calls use _http_request() that handles proxy, SSL verification, etc.
+    For this HelloWorld implementation, no special attributes defined
+    """
+
+    def search_events(
+        self, last_run: dict[str, Any], max_events_per_fetch: int, event_type: str) -> tuple[List[Dict], List[int], str, bool, str]:  # noqa: E501
+        """
+        Searches for GitGuardian alerts using the '/secrets' and '/audit_logs' API endpoints.
+        All the parameters are passed directly to the API as HTTP POST parameters in the request
+
+        Args:
+            last_run (dict): A dict with a key containing the latest event created time we got from last fetch.
+            max_events_per_fetch (int): number of events per fetch
+            get_events (bool, optional): running the function through the get-events command. Defaults to False.
+
+        Returns:
+            List: A list of events that were fetched
+            str: The time to start the next incident fetch.
+        """
+        from_fetch_time = last_run.get("from_fetch_time", "")
+        to_fetch_time = last_run.get("to_fetch_time", "")
+        last_fetched_event_ids = last_run.get("last_fetched_ids", [])
+        next_url_link = last_run.get("next_url_link", "")
+
+        events, last_fetched_event_ids, next_fetch_url, is_nextTrigger = self.retrieve_events(
+            from_fetch_time, to_fetch_time, max_events_per_fetch, last_fetched_event_ids, event_type, next_url_link
+        )
+
+        if is_nextTrigger:
```

Comment:
`nextTrigger` is the name of the field we send to the lastRun object that indicates how many seconds to wait between the fetches.
`is_nextTrigger` at this point doesn't really mean anything, maybe:
```suggestion
        if is_pagination_in_progress:
```